### PR TITLE
263: Use tables for project list

### LINKED ISSUE #263: Standardize on tables for data display
The majority of GlotPress data is displayed in tables, with two exceptions:
1. the projects list
2. the permissions list

These two areas are represented as html ul's.

For consistency we should standardize on tables.

### PR DESCRIPTION
**Pull request by toolstack**
Converts the project list and permissions list to tables.

Also add a few extra columns to the locales page to make it a bit more useful.

Resolves #263.

### CODE REVIEW

**Comment by ocean90** [comment]
Can we have some screenshots please?

**Comment by toolstack** [comment]
Sure...

Locales List:
locales

Projects list:
projects

Permissins list:
permissions

File: assets/css/style.css [hunk 1]
```diff
@@ -189,7 +189,7 @@ textarea {
 	line-height: 1.5em;
 	width: 40em;
 }
-table.translations, table.translation-sets, table.glossary, table.locale-sub-projects {
+table.translations, table.translation-sets, table.glossary, table.locale-sub-projects, table.permissions, table.projects {
```
Comment (GaryJones) [inline assets/css/style.css]: For code standards and readability, have these as one selector per line?

File: gp-templates/locales.php [hunk 2]
```diff
@@ -18,15 +18,21 @@
...
 		<tbody>
 		<?php foreach ( $locales as $locale ) : ?>
 			<tr>
-				<?php echo "<td>" . gp_link_get( gp_url_join( gp_url_current(), $locale->slug ), $locale->english_name ) . "</td>" ?>
-				<?php echo "<td>" . gp_link_get( gp_url_join( gp_url_current(), $locale->slug ), $locale->native_name ) . "</td>" ?>
-				<?php echo "<td>" . gp_link_get( gp_url_join( gp_url_current(), $locale->slug ), $locale->slug ) . "</td>" ?>
+				<td><?php echo gp_link_get( gp_url_join( gp_url_current(), $locale->slug ), $locale->english_name ); ?></td>
+				<td><?php echo gp_link_get( gp_url_join( gp_url_current(), $locale->slug ), $locale->native_name ); ?></td>
+				<td><?php echo gp_link_get( gp_url_join( gp_url_current(), $locale->slug ), $locale->slug ); ?></td>
+				<td><?php 'ltr' == $locale->text_direction ? _e( 'Left to Right', 'glotpress' ) : _e( 'Right to Left', 'glotpress' ); ?></td>
```
Comment (GaryJones) [inline gp-templates/locales.php:32]: Use `===` instead of `==`?

File: assets/css/style.css (hunk 1 above)
Comment (toolstack) [inline assets/css/style.css]: Makes sense.

**Comment by GaryJones** [comment]
For these new tables (and any old tables, but outside the scope of this ticket), I'd like to see `scope` attributes used for `row` and `column` values, and also captions for the whole table. These are easy wins that should increase accessibility. See http://webaim.org/techniques/tables/data for background info.

File: gp-templates/locales.php (hunk 2 above)
Comment (toolstack) [inline gp-templates/locales.php:32]: No need to, simple equals works fine here.

**Comment by toolstack** [comment]
I'd say the `scope` attribute would make sense as a separate PR.

**Comment by GaryJones** [comment]
How about the first `td` in each tbody row being a `th` then?

**Comment by toolstack** [comment]
The `th`'s are in the thead sections already.

**Comment by GaryJones** [comment]
> The th's are in the thead sections already.

Yes, but screen readers can use a `th` in the tbody rows to identify the row. See http://webaim.org/techniques/tables/data#th .

It's a common pattern outside of WP too - see the markup for table examples on http://getbootstrap.com/css/#tables as a random example.

**Comment by toolstack** [comment]
Without the scope attribute that would be very ambiguous.  Could be done at the same time as scope.

**Comment by GaryJones** [comment]
At the end of this, you're creating new tables. It's new markup, and now is the perfect time to get correct markup. Accessibility should not be an add-on later. Other tables can be updated in other PRs, but new markup, IMO, should do what it can do to be accessible from the start.

**Comment by toolstack** [comment]
Rebased against develop and updated to remove the permissions list as it has already been commit via PR  #436.

**Comment by ocean90** [comment]
Issue has been closed as wontfix.

**Comment by GaryJones** [comment]
@ocean90 Won't fix for what reason?

**Pull request closed without merging**